test/e2e/external_gateways.go

```diff
@@ -3595,3 +3624,133 @@ func resetGatewayAnnotations(f *framework.Framework) {
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(), fmt.Sprintf("Delete second external gateway pod %s from ns %s, failed: %v", podName, namespace, err))
+
+	ginkgo.By(fmt.Sprintf("Create second external gateway pod %s from ns %s with readiness probe", podName, namespace))
+	_, err = setupPodWithReadinessProbe(f, podName, nodeSelector, namespace, command, labels)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(), fmt.Sprintf("Create second external gateway pod %s from ns %s with readiness probe, failed: %v", podName, namespace, err))
+	gomega.Eventually(func() bool {
+		var p *corev1.Pod
+		p, err = f.ClientSet.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+		for _, condition := range p.Status.Conditions {
+			if condition.Type == corev1.PodReady {
+				return condition.Status == corev1.ConditionTrue
+			}
+		}
+		return false
+	}).Should(gomega.Equal(true), fmt.Sprintf("Readiness probe for second external gateway pod %s from ns %s, failed: %v", podName, namespace, err))
+}
+
+func handleGatewayPodRemoval(f *framework.Framework, removalType GatewayRemovalType, gatewayPodName, servingNamespace, gatewayIP string, isAnnotated bool) func() {
+	var err error
+	switch removalType {
+	case GatewayDelete:
+		ginkgo.By(fmt.Sprintf("Delete second external gateway pod %s from ns %s", gatewayPodName, servingNamespace))
+		err := deletePodWithWaitByName(context.TODO(), f.ClientSet, gatewayPodName, servingNamespace)
+		framework.ExpectNoError(err, "Delete the gateway pod failed: %v", err)
+		return nil
+	case GatewayUpdate:
+		if isAnnotated {
+			ginkgo.By("Remove second external gateway pod's routing-namespace annotation")
+			annotatePodForGateway(gatewayPodName, servingNamespace, "", gatewayIP, false)
+			return nil
+		}
+
+		ginkgo.By("Updating external gateway pod labels")
+		p := getGatewayPod(f, servingNamespace, gatewayPodName)
+		p.Labels = map[string]string{"name": gatewayPodName}
+		updatePod(f, p)
+		return nil
+	case GatewayDeletionTimestamp:
+		ginkgo.By("Setting finalizer then deleting external gateway pod with grace period to set deletion timestamp")
+		p := getGatewayPod(f, servingNamespace, gatewayPodName)
+		p.Finalizers = append(p.Finalizers, "k8s.ovn.org/external-gw-pod-finalizer")
+		updatePod(f, p)
+		gomega.Eventually(func() bool {
+			p, err = f.ClientSet.CoreV1().Pods(servingNamespace).Get(context.Background(), gatewayPodName, metav1.GetOptions{})
+			if err != nil {
+				return false
+			}
+			return strings.Contains(strings.Join(p.GetFinalizers(), ","), "k8s.ovn.org/external-gw-pod-finalizer")
+		}).Should(gomega.Equal(true), fmt.Sprintf("Update second external gateway pod %s from ns %s with finalizer, failed: %v", gatewayPodName, servingNamespace, err))
+
+		p = getGatewayPod(f, servingNamespace, gatewayPodName)
+		err = e2epod.DeletePodWithGracePeriod(context.Background(), f.ClientSet, p, 1000)
```

Comment:
why 1000? - just to trigger a deletion timestamp being set and ensuring we react to it?